src/ui-test/tests/dev_mode_test.ts

```diff
@@ -0,0 +1,129 @@
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as consts from '../utils/uiTestConstants';
+import { EditorView, SideBarView, VSBrowser, WebDriver } from 'vscode-extension-tester';
+import { devModeIntegration } from '../../IntegrationConstants';
+import {
+	cleanOutputView,
+	outputViewHasText,
+	viewHasItem,
+	updateFileText,
+	contextMenuItemClick
+} from '../utils/waitConditions';
+import {
+	textDoesNotContainAsci,
+	inputBoxQuickPickOrSet,
+	findSectionItem,
+	DoNextTest
+} from '../utils/utils';
+import { assert } from 'chai';
+
+export function devModeTest(extension: string, language: string, doNextTest: DoNextTest) {
+
+	describe(`Dev Mode test with the integration file update`, function () {
+		this.timeout(60000);
+
+		let driver: WebDriver;
+
+		const [initialCodeMessage, updatedCodeMessage, initialLogMessage, updatedLogMessage] = consts.prepareCodeLogMessages(extension, language);
+
+		before(async function () {
+			driver = VSBrowser.instance.driver;
+		});
+
+		after(async function () {
+			this.timeout(60000);
+			await new EditorView().closeAllEditors();
+			const item = await findSectionItem(consts.extensionName, consts.integrationFileName.toLowerCase());
+			if (item !== undefined) {
+				await driver.wait(() => { return contextMenuItemClick(item, consts.integrationRemove); });
+				const content = new SideBarView().getContent();
+				await driver.wait(async () => {
+					return viewHasItem(content, consts.extensionName, consts.integrationFileName.toLowerCase())
+						.then(val => {
+							return !val
+						});
+				});
+			}
+			await cleanOutputView();
+		});
+
+		beforeEach(function () {
+			if (!doNextTest.doNextTest) {
+				this.skip();
+			}
+		});
+
+		afterEach(function () {
+			if (this.currentTest?.state === 'failed' && this.id !== 'independent') {
+				doNextTest.stopTest();
+			}
+		});
+
+		it(`Select ${consts.startIntegration} in the popup menu`, async function () {
+			this.timeout(30000);
+			const item = await findSectionItem(consts.testFolder, `${consts.integrationFileName}.${extension}`);
+			await driver.wait(() => { return contextMenuItemClick(item, consts.startIntegration); });
+		});
+
+		it(`Start integration with '${devModeIntegration}' command`, async function () {
+			this.timeout(5000);
+			assert.isTrue(await inputBoxQuickPickOrSet('pick', devModeIntegration));
+		});
+
+		it(`Initial integration pod started`, async function () {
+			this.timeout(300000);
```

Comment:
might be worthy to use 2 different constants for the allowed tiemout. One is 30000 so 30 seconds and anlther one is 300000 so 5 minutes. But hard to notice the difference on first sight as there is just one zero difference in a "long" list of zero

Reply:
I think both are the same 5 minutes

Reply:
there are (at least) 2 `300000` so 5 minute:
- https://github.com/camel-tooling/vscode-camelk/pull/1200/files#diff-d508539477f02a8178000d6ede0f22f9d4920d04a615300f765c3b0dfda70732R92
- https://github.com/camel-tooling/vscode-camelk/pull/1200/files#diff-4aa4e98b91465eca8010a71539b096e845c2cb34cc4d04668757ddc44dae0b0dR91

and several `30000` so 30 seconds:
- https://github.com/camel-tooling/vscode-camelk/pull/1200/files#diff-4aa4e98b91465eca8010a71539b096e845c2cb34cc4d04668757ddc44dae0b0dR96
- https://github.com/camel-tooling/vscode-camelk/pull/1200/files#diff-4aa4e98b91465eca8010a71539b096e845c2cb34cc4d04668757ddc44dae0b0dR101
- https://github.com/camel-tooling/vscode-camelk/pull/1200/files#diff-4aa4e98b91465eca8010a71539b096e845c2cb34cc4d04668757ddc44dae0b0dR107

Reply:
I have prepared timeouts based on QE pipeline (with OpenStack and openshift)
Today I was able to optimize it a lot.
I will check tests tomorrow to reduce at least 300,000
There are already many tests with the same timeouts, I will prepare some constants.

Reply:
actually these timeouts are setting the maximum time so probably it will be always resolved more soon
In my opinion it can only bring more stability to downstream jenkins pipelines where we are testing multiple architectures

Reply:
Yes, the biggest value that was noticed by me was something +-478,000 for the first `Initial integration pod started`
But now after all updates, I hope I can significantly reduce it (up to 30.000 in case of network problems, during standard behaviour now <10.000 in downstream)

Reply:
Done